lib/stdlib/test/re_SUITE.erl

```diff
@@ -1115,3 +1115,44 @@ error_info(_Config) ->
          {urun, 3}                              %Internal.
         ],
     error_info_lib:test_error_info(re, L).
+
+pattern_is_sub_binary(Config) when is_list(Config) ->
+    %% Aligned sub binary - will not copy the binary
+    Bin = <<"pattern = ^((:|(0?|([1-9a-f][0-9a-f]{0,3}))):)((0?|([1-9a-f][0-9a-f]{0,3})):){0,6}(:|(0?|([1-9a-f][0-9a-f]{0,3})))$">>,
+    Subject = <<"::1">>,
+    {_,RE} = split_binary(Bin, 10),
+    {ok,REC} = re:compile(RE),
+    match = re:run(Subject, REC, [{capture, none}]),
+    match = re:run(Subject, RE, [{capture, none}]),
+    nomatch = re:run(Subject, Bin, [{capture, none}]),
+    %% Unaligned sub binary - will result in a copy operation
+    <<0:1, RE2/binary>> = Bin2 = <<0:1, "^((:|(0?|([1-9a-f][0-9a-f]{0,3}))):)((0?|([1-9a-f][0-9a-f]{0,3})):){0,6}(:|(0?|([1-9a-f][0-9a-f]{0,3})))$">>,
+    {ok,REC2} = re:compile(RE2),
+    match = re:run(Subject, REC2, [{capture, none}]),
+    match = re:run(Subject, RE2, [{capture, none}]),
+    try
+        nomatch = re:run(Subject, Bin2, [{capture, none}])
+    catch error:badarg ->
+        %% *** argument 2: neither an iodata term nor a compiled regular expression
+        ok
+    end.
```

Comment:
Should this be something like
`ok = try ... catch error:badarg -> ok end.`
to make sure it actually raised a badarg.